src/Extensions/ElementalAreasExtension.php

```diff
@@ -168,9 +178,12 @@ public function updateCMSFields(FieldList $fields)
             return;
         }
 
-        // add an empty holder for content as some module explicitly use insert
-        // after content.
-        $fields->replaceField('Content', new LiteralField('Content', ''));
+        // add an empty holder for content as some module explicitly use insert after content
+        $globalReplace = !Config::inst()->get(self::class, 'keep_content_fields');
+        $classOverride = Config::inst()->get(get_class($this->owner), 'elemental_keep_content_field');
```

Comment:
The only minor feedback I have is that I'd call this just `keep_content_field`. You could actually probably just call both config settings `keep_content_field` - the plural doesn't really make much sense unless it would affect multiple content fields.

Reply:
The global one would affect multiple fields, was my thought there. But not on the same model object, yeah.

I used the prefix for the singular option to indicate it's meaning when assigned to an otherwise arbitrary model - a sort of namespace I guess. It has irked me in the past when I've looked up a model on a project to see a whole bunch of config values that I have no idea where they're coming from (particularly where multiple `DataExtension`s are involved), so I'd like to give a developer at least some indication in what it is for (rather than having to search a code base to see where the config value is referenced).

I also like the distinction between "this affects everything" and "this affects _this_ thing" when using the config key in conversation. I think it could reduce confusion and cognitive load.
But your counter argument works equally as well in regards to adding cognitive load with different names, I guess.

Do you feel strongly about this?

Reply:
> Do you feel strongly about this?

No.

Reply:
Cool, what are your thoughts on my response? :)
Does it make sense, should I add comments? Sounds good, YOLO?